tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContactSearchEngineContract.scala

```diff
@@ -649,4 +651,47 @@ trait EmailAddressContactSearchEngineContract {
     assertThat(SMono.fromPublisher(testee().get(domain, mailAddressA)).asJava().map(_.fields).block())
       .isEqualTo(contactFieldsA)
   }
+
+  @Test
+  def shouldReturnOnlyEmailMatchWhenPartHasAtSign(): Unit = {
+    val mailAddress: MailAddress = new MailAddress("loic.sineau@example.com")
+    val contactFields: ContactFields = ContactFields(mailAddress, "SINEAU", "Loic")
+    SMono(testee().index(accountId, contactFields)).block()
+    SMono(testee().index(domain, contactFields)).block()
+
+    awaitDocumentsIndexed(MatchAllQuery(), 2)
+
+    val partHasAtSign: String = "anyone@loic"
+
+    assertThat(SFlux.fromPublisher(testee().autoComplete(accountId, partHasAtSign))
+      .map(_.fields.address)
+      .collectSeq()
+      .block().asJava)
+      .doesNotContain(mailAddress)
+  }
+
+  @Test
+  def shouldReturnOnlyPrefixedByUserInput(): Unit = {
+    val mailAddress: MailAddress = new MailAddress("bob@linagora.com")
+    val contactFields: ContactFields = ContactFields(mailAddress, "Bin", "Zet")
+    SMono(testee().index(accountId, contactFields)).block()
+    SMono(testee().index(domain, contactFields)).block()
+
+    val whatEverMailAddress: MailAddress = new MailAddress("whatever@linagora.com")
+    val whatEverContactFields: ContactFields = ContactFields(whatEverMailAddress, "What", "Ever")
+
+    SMono(testee().index(accountId, whatEverContactFields)).block()
+    SMono(testee().index(domain, whatEverContactFields)).block()
+
+    awaitDocumentsIndexed(MatchAllQuery(), 4)
+
+    val prefixedPath: String = "bob@linagora.c"
+
+    assertThat(SFlux.fromPublisher(testee().autoComplete(accountId, prefixedPath))
+      .map(_.fields.address)
+      .collectSeq()
+      .block().asJava)
+      .containsExactlyInAnyOrder(mailAddress)
+      .doesNotContain(whatEverMailAddress)
+  }
```

Comment:
I suggest a more realistic test case - which I have today:
![image](https://github.com/linagora/tmail-backend/assets/55171818/676196d8-c8dc-488b-902e-dadef5dbf4c5)

And yes with the fix in this PR, it will be fixed.

Reply:
```suggestion
  @Test
  def shouldReturnOnlyPrefixedByUserInput(): Unit = {
    val mailAddress: MailAddress = new MailAddress("hqtran@linagora.com")
    val contactFields: ContactFields = ContactFields(mailAddress, "Hong Quan", "Tran")
    SMono(testee().index(accountId, contactFields)).block()
    SMono(testee().index(domain, contactFields)).block()

    val whatEverMailAddress: MailAddress = new MailAddress("lgs-openpaas-dev@linagora.com")
    val whatEverContactFields: ContactFields = ContactFields(whatEverMailAddress, "lgs-openpaas-dev@linagora.com", "")

    SMono(testee().index(accountId, whatEverContactFields)).block()
    SMono(testee().index(domain, whatEverContactFields)).block()

    awaitDocumentsIndexed(MatchAllQuery(), 4)

    val prefixedPath: String = "hqtran@linagora.c"

    assertThat(SFlux.fromPublisher(testee().autoComplete(accountId, prefixedPath))
      .map(_.fields.address)
      .collectSeq()
      .block().asJava)
      .containsExactlyInAnyOrder(mailAddress)
      .doesNotContain(whatEverMailAddress)
  }
```

Reply:
I think we shouldn't use "real email" in the source code.

Reply:
Let's change it a bit diff I guess.

Reply:
Or if your test failed before and now passed with this fix. I am fine.